src/smt/optimization_solver.cpp

```diff
@@ -28,6 +30,69 @@ using namespace cvc5::omt;
 namespace cvc5 {
 namespace smt {
 
+std::ostream& operator<<(std::ostream& out, const OptimizationResult& result)
+{
+  // check the output language first
+  OutputLanguage lang = language::SetLanguage::getLanguage(out);
+  if ((!language::isOutputLang_smt2(lang))
+      && (!language::isOutputLangSygus(lang)))
+  {
+    Unimplemented()
+        << "Only SMTLib2 or Sygus languages support optimization right now";
+  }
+  out << "(" << result.getResult();
+  switch (result.getResult().isSat())
+  {
+    case Result::SAT: CVC5_FALLTHROUGH;
```

Comment:
[This](https://en.cppreference.com/w/cpp/language/attributes/fallthrough#:~:text=A%20fallthrough%20statement%20may%20only,same%20iteration%20of%20that%20loop.) suggests that the `fallthrough` attribute relates to a previous case, but this is the first case. Do you get a compiler warning if you remove this?

Reply:
I haven't tried. But I got a warning when I forgot to add a break at another switch statement.

Reply:
removing `[[fallthrough]]` seems to work. I got a warning because the case I was working on has some code there instead of direct falling through. but I would prefer keeping `CVC5_FALLTHROUGH`, in case you use an older standard like C++0x, you could just make the macro to be empty.

Reply:
Well as it is right now, it is in the first case, while the documentation of `fallthrough` only mentions how it is used when there are cases before it. Hence it is not clear to me how this is useful here. Please remove it. If you do find a good reason to keep it, which is consistent with `fallthrough` documentation, please document it in the code and we can iterate over it.

Reply:
I think you misunderstood cppreference. It means: the case using [[fallthrough]] is the "previous case", while the case following it is the "next case". Your understanding might be: the case using [[fallthrough]] is "current case", the case before that is "previous case" and the following "current case" is the "next case". Well... I don't see any reason why it forbids using this in the first case. check proposal P0188R1 here: https://gcc.gnu.org/projects/cxx-status.html

Reply:
OK, removing [[fallthrough]] looks a bit cleaner. I removed it purely because of aesthetic issue.